shopelectro/tests/tests_models.py

```diff
@@ -21,3 +24,42 @@ def test_creation_deactivated_product(self):
             ))
         except Exception as error:
             self.fail(f'Creation of existing product failed: {{ error }}')
+
+
+class TagModel(TestCase):
+
+    fixtures = ['dump.json']
+
+    def get_products(self):
+        return Product.objects.all()[:3]
+
+    def test_get_brands_content(self):
+        brand_group = TagGroup.objects.get(name=settings.BRAND_TAG_GROUP_NAME)
+        for product, tag in Tag.objects.get_brands(self.get_products()).items():
+            self.assertEquals(tag.group, brand_group)
+            self.assertIn(product, tag.products.all())
+
+    def test_filter_by_products_equality(self):
+        products = self.get_products()
+        self.assertEquals(
+            set(Tag.objects.filter_by_products(products)),
+            set(chain.from_iterable(product.tags.all() for product in products)),
+        )
+
+    def test_filter_by_products_quantity(self):
+        products = self.get_products()
+        self.assertEquals(
+            Tag.objects.filter_by_products(products).count(),
+            len(set(chain.from_iterable(product.tags.all() for product in products))),
+        )
+
+
+class QueriesNumber(TransactionTestCase):
```

Comment:
this class has the semantical intersection with `TagModel`.
It's not clear where we should put `test_tag_save` method: to this class or to `TagModel`.

I propose to call it `TagQuerySet`

Reply:
I am going to create docstring for uit